File: LayoutTests/http/tests/navigation/resources/save-ping-and-redirect-to-save-ping.php
<?php
require_once 'ping-file-path.php';

$DO_NOT_CLEAR_COOKIES = true; // Used by save-Ping.php
require_once 'save-Ping.php';

header('HTTP/1.1 307');
header('Location: save-Ping.php' . (!empty($_SERVER['QUERY_STRING']) ? '?' . $_SERVER['QUERY_STRING'] : ''));
?>

File: LayoutTests/http/tests/navigation/resources/save-ping-and-redirect-to-save-ping.py
#!/usr/bin/env python3

import os
import sys
from save_ping import save_ping

query_string = os.environ.get('QUERY_STRING', '')
if query_string != '':
    query_string = '?{}'.format(query_string)

sys.stdout.write(
    'status: 307\r\n'
    'Location: save-ping.py{}\r\n'.format(query_string)
)

save_ping(True)
